### PR TITLE
Fix #3656 Improve camel-quarkus-paho-mqtt5 test coverage

### LINKED ISSUE #3656: Improve camel-quarkus-paho-mqtt5 test coverage
- MQTT over WebSocket test
- FilePersistence test
- Override the default topic and set to a dynamic topic
- Reconnect after failure
- Security

### PR DESCRIPTION
**Pull request by zhfeng**

### CODE REVIEW

File: integration-tests/paho-mqtt5/src/main/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5Resource.java [hunk 1]
```diff
@@ -60,12 +86,94 @@ public Response producePahoMessage(
...
+
+        System.setProperty("javax.net.ssl.keyStore", keystore);
+        System.setProperty("javax.net.ssl.keyStorePassword", password);
+        System.setProperty("javax.net.ssl.trustStore", keystore);
+        System.setProperty("javax.net.ssl.trustStorePassword", password);
+    }
+
+    private void removeKeyStore(String keystore) {
+        try {
+            Files.delete(Paths.get(keystore));
```
Comment (jamesnetherton) [inline integration-tests/paho-mqtt5/src/main/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5Resource.java:175]: Should we unset the SSL system properties here?

Comment (zhfeng) [reply integration-tests/paho-mqtt5/src/main/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5Resource.java:175]: good catch! I will unset these properties.

Comment (jamesnetherton) [reply integration-tests/paho-mqtt5/src/main/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5Resource.java:175]: Also, I guess we want this to always happen. E.g even if the call to the `ProducerTemplate` throws an exception.

File: integration-tests/paho-mqtt5/src/test/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5TestResource.java [hunk 2]
```diff
@@ -32,25 +36,76 @@
...
     public Map<String, String> start() {
         LOGGER.info(TestcontainersConfiguration.getInstance().toString());
 
         try {
-            container = new GenericContainer<>(IMAGE)
-                    .withExposedPorts(TCP_PORT)
-                    .withLogConsumer(new Slf4jLogConsumer(LOGGER))
+            Map<String, String> result = new HashMap<>();
+
+            if (useFixedPort) {
```
Comment (aldettinger) [inline integration-tests/paho-mqtt5/src/test/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5TestResource.java:72]: Could you please describe why we need to have use `useFixedPort` ? When should we use it ? And when not to use it please ?

Comment (aldettinger) [reply integration-tests/paho-mqtt5/src/test/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5TestResource.java:72]: Ok, So it's only for manual testing. And yes, I know this re-connection scenario is hard to setup :) Many thanks.

**Review by aldettinger — APPROVED** [review]

**Review by jamesnetherton — APPROVED** [review]

**Pull request merged**